bundle-httpcache/src/main/java/com/adobe/acs/commons/httpcache/config/impl/GroupHttpCacheConfigExtension.java

```diff
@@ -2,31 +2,54 @@
 
 import com.adobe.acs.commons.httpcache.config.HttpCacheConfig;
 import com.adobe.acs.commons.httpcache.config.HttpCacheConfigExtension;
-import com.adobe.acs.commons.httpcache.exception.HttpCacheReposityAccessException;
+import com.adobe.acs.commons.httpcache.exception.HttpCacheKeyCreationException;
+import com.adobe.acs.commons.httpcache.exception.HttpCacheRepositoryAccessException;
+import com.adobe.acs.commons.httpcache.keys.CacheKey;
+import com.adobe.acs.commons.httpcache.keys.CacheKeyFactory;
 import com.adobe.acs.commons.httpcache.util.UserUtils;
 import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.apache.felix.scr.annotations.Activate;
 import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Modified;
+import org.apache.felix.scr.annotations.Property;
+import org.apache.felix.scr.annotations.PropertyUnbounded;
 import org.apache.felix.scr.annotations.Service;
 import org.apache.jackrabbit.api.security.user.User;
 import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.commons.osgi.PropertiesUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.jcr.RepositoryException;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
 
 /**
  * This cache config extension assumes that custom attributes are aem group names. It accepts the http request only if
  * at least one of the configured groups is present in the request user's group membership list.
  */
-@Component(immediate = true)
+@Component(immediate = true, metatype = true)
 @Service
-public class GroupHttpCacheConfigExtension implements HttpCacheConfigExtension {
+public class GroupHttpCacheConfigExtension implements HttpCacheConfigExtension, CacheKeyFactory {
     private static final Logger log = LoggerFactory.getLogger(GroupHttpCacheConfigExtension.class);
 
+    // Custom cache config attributes
```

Comment:
I made this config factory and move along with cacheconfig (n - n mapping). Result is so beautiful. For the same uri, i was able to cache separately for 2 different groups.

<img width="1512" alt="cache-config-geometrix" src="https://cloud.githubusercontent.com/assets/7424268/11422636/a4d16cd4-9463-11e5-8d74-dc7dfb638d70.png">
<img width="1514" alt="cache-config-admin" src="https://cloud.githubusercontent.com/assets/7424268/11422637/a4d38316-9463-11e5-9374-039906a2242c.png">

sling:OSGi config nodes for GroupHttpCacheConfigExtension
com.adobe.acs.commons.httpcache.config.impl.GroupHttpCacheConfigExtension-admins.xml
<?xml version="1.0" encoding="UTF-8"?>
<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
          jcr:primaryType="sling:OsgiConfig"
          config.name="group-admins"
          httpcache.config.extension.user-groups="administrators"/>

com.adobe.acs.commons.httpcache.config.impl.GroupHttpCacheConfigExtension-geo.xml
<?xml version="1.0" encoding="UTF-8"?>
<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
          jcr:primaryType="sling:OsgiConfig"
          config.name="group-geo"
          httpcache.config.extension.user-groups="geometrixx-twitter"/>